src/Scanner/Scanner.cpp

```diff
@@ -0,0 +1,421 @@
+            char c = advance();
+            switch (c) {
+                case '-': 
+                    if (match('=')) {
+                        addToken(m_tokenFactory->createToken("-=", m_line,TokenType::SUBEQ));
+                    }
+                    else if(match('-')){
+                        addToken(m_tokenFactory->createToken("--", m_line,TokenType::DEC));
+                    }
+                    else if (match('>')) {
+                        addToken(m_tokenFactory->createToken("->", m_line,TokenType::ARROW));
+                    }
+                    else {
+                        addToken(m_tokenFactory->createToken("-", m_line,TokenType::SUB)); 
+                    }
+                    break;
+                case '+': 
+                    if (match('=')){
+                        addToken(m_tokenFactory->createToken("+=", m_line,TokenType::ADDEQ));
+                    }
+                    else if (match('+')){
+                        addToken(m_tokenFactory->createToken("++", m_line,TokenType::INC));
+                    }
+                    else {
+                        addToken(m_tokenFactory->createToken("+", m_line,TokenType::ADD));
+                        }
+                    break;
+                case '*': 
+                    if (match('=')){
+                        addToken(m_tokenFactory->createToken("*=", m_line,TokenType::MULEQ));
+                    }
+                    else {
+                        addToken(m_tokenFactory->createToken("*", m_line,TokenType::MUL));
+                    }
+                    break;
+                /*indicating string*/
+                case '"':
+                    addToken(m_tokenFactory->createToken(getString(), m_line,TokenType::STRING));
+                    break;
+                case '|':
+                    if (match('|')){
+                        addToken(m_tokenFactory->createToken("||", m_line,TokenType::OR));
+                    }
+                    else if (match('=')) {
+                        addToken(m_tokenFactory->createToken("|=", m_line, TokenType::OREQ));
+                    }
+                    else{
+                    addToken(m_tokenFactory->createToken("|", m_line,TokenType::BITOR));
+                    }
+                    break;
+                case '&':
+                    if (match('&')){
+                        addToken(m_tokenFactory->createToken("&&", m_line,TokenType::AND));
+                    }
+                    else if (match('=')) {
+                        addToken(m_tokenFactory->createToken("&=", m_line, TokenType::ANDEQ));
+                    }
+                    else{
+                    addToken(m_tokenFactory->createToken("&", m_line,TokenType::BITAND));
+                    }
+                    break;
+                case '^':
+                    if (match('=')) {
+                        addToken(m_tokenFactory->createToken("^=", m_line, TokenType::XOREQ));
```

Comment:
lets try to register the lexems statically and getToken instead of creating it dynamically. I tried to demonstrate my point of the implementation, you can find it here: https://godbolt.org/z/f1WGrMbvo